app/models/pool.ts

```diff
@@ -70,47 +41,70 @@ export interface PoolAttributes {
 /**
  * Class for displaying Batch pool information.
  */
-export class Pool extends PoolRecord {
+@Model()
+export class Pool extends Record<PoolAttributes> {
+    @Prop()
     public allocationState: string;
+    @Prop()
```

Comment:
Don't suppose you would be keen to add a space between the property and the next @Prop()? It makes it pretty hard to read when there are no spaces between properties. I am sure this should violate a style guide somewhere :) well it would in C# anyway.